src/components/deviceprofileBuilder.ts

```diff
@@ -179,89 +177,142 @@ function getCodecProfiles(): CodecProfile[] {
-    const h265Profile: string = getH265ProfileSupport(currentDeviceId);
-
-    const h265Conditions: CodecProfile = {
-        Codec: 'h265',
-        Conditions: [
-            createProfileCondition(
-                ProfileConditionValue.IsAnamorphic,
-                ProfileConditionType.NotEquals,
-                'true'
-            ),
-            createProfileCondition(
-                ProfileConditionValue.VideoProfile,
-                ProfileConditionType.EqualsAny,
-                h265Profile
-            ),
-            createProfileCondition(
-                ProfileConditionValue.VideoLevel,
-                ProfileConditionType.LessThanEqual,
-                h265Level.toString()
-            ),
-            createProfileCondition(
-                ProfileConditionValue.Width,
-                ProfileConditionType.LessThanEqual,
-                maxWidth.toString(),
-                true
-            )
-        ],
-        Type: CodecType.Video
-    };
-
-    CodecProfiles.push(h265Conditions);
+        if (videoProfiles.length === 0) {
+            continue;
+        }
 
-    // the following are codec independent conditions so hold for all codecs
-    const videoConditions: CodecProfile = {
-        Conditions: [
-            createProfileCondition(
-                ProfileConditionValue.Width,
-                ProfileConditionType.LessThanEqual,
-                getMaxWidthSupport(currentDeviceId).toString(),
-                true
-            )
-        ],
-        Type: CodecType.Video
-    };
+        const maxLevels: number[] = [];
+        const maxBitDepths: number[] = [];
+        const maxResolutions: Resolution[] = [];
+
+        for (const videoProfile of videoProfiles) {
+            const maxVideoLevel =
+                getVideoCodecHighestLevelSupport(videoCodec, videoProfile) ?? 0;
+
+            const maxBitDepth =
+                getVideoCodecHighestBitDepthSupport(
+                    videoCodec,
+                    videoProfile,
+                    maxVideoLevel
+                ) ?? 0;
+
+            const maxResolution = getMaxResolutionSupported(
+                videoCodec,
+                videoProfile,
+                maxVideoLevel,
+                maxBitDepth
+            );
+
+            maxLevels.push(maxVideoLevel);
+            maxBitDepths.push(maxBitDepth);
+            maxResolutions.push(maxResolution);
+        }
 
-    CodecProfiles.push(videoConditions);
+        // If all other constraints are equal, merge into one condition. This
+        // is pretty common.
+        if (
+            maxLevels.every((l) => l === maxLevels[0]) &&
+            maxBitDepths.every((b) => b === maxBitDepths[0]) &&
+            maxResolutions.every((r) => r.equals(maxResolutions[0]))
+        ) {
+            const maxLevel = maxLevels[0];
+            const maxBitDepth = maxBitDepths[0];
+            const maxResolution = maxResolutions[0];
+
+            const profileConditions = [
```

Comment:
You didn't change the logic from before, but this condition seems odd... why do we care if it is anamorphic?

Reply:
I don't know enough about what videos are anamorphic anymore (iiuc, it's an artifact of 35mm film -> old CRT 4:3 home displays), so I didn't really change it here. If you have some context as to why we won't care about this anymore, more than happy to remove it.